src/lua/crypto_schnorr_signature.lua

```diff
@@ -152,18 +155,18 @@ function schnorr.verify(pk, m, sig)
    if iszen(type(pk)) then pk = pk:octet() end
    --the follwing "lifts" pk to an ECP with x = pk and y is even
    local P = ECP.new(BIG.new(pk))    
-   assert(P, "lifting failed")
+   if not _schnorr_assert(P, "lifting failed") then return false end
```

Comment:
how about `return _schnorr_assert(...)`

Reply:
you want to exit only in the case _schnorr_assert is false, in the case it is true yo want to continue with all the checks